invenio_rdm_records/services/communities/service.py

```diff
@@ -37,10 +52,107 @@ def record_cls(self):
         """Factory for creating a record class."""
         return self.config.record_cls
 
+    def _exists(self, identity, community_id, record):
+        """Return the request id if an open request already exists, else None."""
+        results = current_requests_service.search(
```

Comment:
moderate: I now understand what you meant the other about depending on ES for the check. Would it be possible to make this check certain (irrelevant of the automatic index refresh), by waiting for the refresh when indexing these requests? (i.e. [`?refresh=wait_for`](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html))

Although I have a feeling that the `RecordIndexOp` is deeper in the requests service and it won't be easy to control/pass this easily...

Reply:
Thanks for the suggestion, I will look into that. As first implementation, I checked if I could use DB instead, but no search possible... I am not super happy about ES search, but I don't see any other easy solution.
I will check the `refresh` param.